Documentation/cmdref/cilium-agent.md

```diff
@@ -259,6 +260,7 @@ cilium-agent [flags]
       --trace-payloadlen int                                    Length of payload to capture when tracing (default 128)
   -t, --tunnel string                                           Tunnel mode {vxlan, geneve, disabled} (default "vxlan" for the "veth" datapath mode)
       --tunnel-port int                                         Tunnel port (default 8472 for "vxlan" and 6081 for "geneve")
+      --tunnel-protocol string                                  Encapsulation protocol to use for the overlay ("vxlan" or "geneve") (default "vxlan")
```

Comment:
Do we want to note that this setting is either incompatible with or overrides the other flag added "--enable-native-routing"?

Reply:
hmm, okay. My concern is that, what you just mentioned above isn't exactly clear from the CLI documentation. 
Is it worth mentioning this? I do understand that you may not be able to encapsulate all effects of a flag in its short doc string, but I'm not quite sure where that limit exists.

Reply:
- `cilium-agent --tunnel-protocol=vxlan` => tunneling mode (default), with VXLAN (default).
- `cilium-agent --enable-native-routing=false --tunnel-protocol=vxlan` => same.
- `cilium-agent --enable-native-routing --tunnel-protocol=vxlan` => not sure which guide you're following, get your shit together. Will ignore the `--tunnel-protocol` flag.
- `cilium-agent --enable-native-routing --enable-egress-gateway --tunnel-protocol=vxlan` => will enable egress gateway and use VXLAN for the ad-hoc tunnel.

Reply:
I think this is nitpicking and it doesn't fix any real issue (we've always assumed users follow guides and don't try flags at random), but I like the flag, so I'll change the PR to use it.

Reply:
> we've always assumed users follow guides

Sure, but IMHO if a flag name can be chosen such that it's a self-descriptive, then we should aim for it.

Reply:
lol, tbh, i liked where this original pr was going, however without the caveat or the egress gateway tunnel.

not to throw another suggestion in but, in my head it makes sense to not overload any of the flags.

```
--enable-native-routing // ignores the --tunnel-protocol flag all together
--tunnel-protocol={vxlan,geneve} // ignores --enable-native-routing flag all together
--enable-egress-gateway // enables egress gateway, independent of "main" routing mode
--egress-gateway-tunnel-mode // sets the tunnel mode for the egress gateway ad-hoc tunnels
```

each flag is well defined for its purpose.